### PR TITLE
Document config overriding functionality

### PR DESCRIPTION
**Pull request by poslegm**
Scalafmt has powerful config overriding feature but nobody knows about it :(

Helps with #1534

### CODE REVIEW

**Comment by poslegm** [comment]
Is there reasons to hide it?

File: docs/configuration.md [hunk 1]
```diff
@@ -13,6 +13,19 @@ align = more    // For pretty alignment.
 maxColumn = 100 // For my wide 30" display.
 ```
 
+There is a possibility to override scalafmt config for a specific file with
+`// scalafmt: {}` comment:
+
+```scala mdoc:scalafmt
+---
+// inside build.sbt
+// scalafmt: { align = most }
```
Comment (kitbellew) [inline docs/configuration.md]: does it accept multiple parameters?

File: docs/configuration.md [hunk 2]
```diff
@@ -13,6 +13,19 @@ align = more    // For pretty alignment.
 maxColumn = 100 // For my wide 30" display.
 ```
 
+There is a possibility to override scalafmt config for a specific file with
```
Comment (kitbellew) [inline docs/configuration.md]: it's not really file, it's group of lines. i think you can use another comment further down, to modify again.

unfortunately, there's no `off` setting; if you want to restore, you need to set it explicitly.

File: docs/configuration.md (hunk 1 above)
Comment (poslegm) [reply docs/configuration.md]: yes, added another param into doc

File: docs/configuration.md (hunk 2 above)
Comment (poslegm) [reply docs/configuration.md]: specified it

Comment (olafurpg) [reply docs/configuration.md]: It should be fairly easy to implement some kind of `off` setting to revert to the original settings

**Pull request merged**